src/daft-connect/src/display.rs

```diff
@@ -0,0 +1,382 @@
+use std::fmt::Write;
+
+use daft_core::prelude::*;
+
+pub trait SparkDisplay {
+    fn repr_spark_string(&self) -> String;
+}
```

Comment:
Minor nit, but usually if a trait is only implemented for a set number of types, using an enum to achieve the same behaviour is a better choice.

I.e.:

```rs
enum SparkDisplay {
  Schema(Schema),
  Field(Field),
  DataType(DataType),
}

impl SparkDisplay {
  fn repr_spark_string(&self) -> String {
    match self {
      Self::Schema(..) => ..,
      Self::Field(..) => ..,
      Self::DataType(dt) => type_to_string(dt),
    }
  }
}
```

Reply:
so this pattern closely resembles what we do with the other `repr_` methods. such as [repr_mermaid](https://github.com/universalmind303/Daft/blob/e149dcd75ca7530dceb7bbdc6d1e6f6e69b60dfa/src/common/display/src/mermaid.rs#L7)